### PR TITLE
Update batchflow

### PR DESCRIPTION
**Pull request by SergeyTsimfer**
Update `batchflow` to the latest (and greatest) version

Also make `plot_image/overlap` work with arbitrary colors

### CODE REVIEW

File: seismiqb/src/crop_batch.py [hunk 1]
```diff
@@ -535,10 +536,9 @@ def masks_to_horizons(self, ix, src='masks', src_locations='locations', dst='pre
         _ = dst, mean_threshold, adjacency, skip_merge
 
         # Threshold the mask, transpose and rotate the mask if needed
-        pos = self.get_pos(None, src, ix)
-        mask = getattr(self, src)[pos]
+        mask = self.get(ix, src)
         if np.array(order).reshape(-1, 3).shape[0] > 0:
-            order = order[pos]
+            order = self.get(ix, order)
```
Comment (dmylzenova) [inline seismiqb/src/crop_batch.py:541]:
```suggestion
            order = self.get(ix, 'order')
```

File: seismiqb/src/controllers/base.py [hunk 2]
```diff
@@ -295,15 +296,17 @@ def train(self, dataset, model_config=None, device=None, n_iters=300,
                   savepath=self.make_save_path('model_loss.png'))
 
         self.model_pipeline = model_pipeline
-
         last_loss = np.mean(model_pipeline.v('loss_history')[-50:])
         self.log(f'Train finished; last loss is {last_loss}')
-        self.log(f'Cache size: {[len(item._cached_load.cache()) for item in dataset.geometries.values()]}')
+        self.log(f'Cache sizes: {[item.cache_size for item in dataset.geometries.values()]}')
```
Comment (dmylzenova) [inline seismiqb/src/controllers/base.py:301]: Is this all that is being cached?

File: seismiqb/src/utils.py [hunk 3]
```diff
@@ -575,3 +576,49 @@ def nb_mode(array, mask):
...
+        high = low + self.shifts_edges
+        return self.state_sampler(low=low, high=high)
+
+@njit
+def generate_points(edges, divisors, lengths, indices):
+    """ Accelerate sampling method of `HorizonSampler`. """
+    low = np.zeros((len(indices), len(lengths)))
+
+    for i, idx in enumerate(indices):
+        idx_copy = idx
```
Comment (dmylzenova) [inline seismiqb/src/utils.py:618]: Is it necessary? It is a duplicate of line 620?

File: seismiqb/src/controllers/base.py (hunk 2 above)
Comment (SergeyTsimfer) [reply seismiqb/src/controllers/base.py:301]: That is everything that is cached with `lru_cache` mechanism

**Pull request merged**